src/Frame.h

```diff
@@ -316,6 +325,13 @@ class Frame :  public Obj {
 	/** Associates ID's offsets with values. */
 	std::unique_ptr<Element[]> frame;
 
+	/**
+	 * The offset we're currently using for references into the frame.
+	 * This is how we support inlined functions without having to
+	 * alter the offsets associated with their local variables.
+	 */
+	int current_offset;
```

Comment:
Was hard to tell whether there's some missing applications of this new offset.

Example: calling `SetElement(n, v)` and then `GetElement(n)` seems like it may not actually return a reference to the `v` that was just set at index `n` when offset is non-zero.

Idea for improvement: wrap the `frame` member in a struct w/ overloaded `operator[]` that ensures the offset gets applied to all accesses?

Reply:
I'm pretty sure there aren't any missing applications, as they'd fail immediately (running the test suite with inlining will present zillions of such opportunities).  The one you flagged is technically an issue but in practice not, as the only time `GetElement` is called with an numeric index rather than an `ID*` is during function call invocation, and those can't occur with non-zero `current_offset`.

If you want this to be tidier, eems simpler to make that call stand out a bit (e.g., `GetElementUsingIndex`) and adjust it to add in the offset.